test/emails/post-emails.api-test.ts

```diff
@@ -0,0 +1,220 @@
+
+  it('calls NotifyClient.sendEmail with custom reference', async () => {
+    const customReference = valueGenerator.string({ length: 10 });
+
+    await api.post(mdmPath, [{ ...request[0], reference: customReference }], { govUkNotifyKey });
+
+    expect(sendEmailMethodMock).toHaveBeenCalledWith(request[0].templateId, request[0].sendToEmailAddress, {
+      personalisation: request[0].personalisation,
+      reference: customReference,
+    });
+  });
+
+  it('returns a 400 response if header `govUkNotifyKey` is missing', async () => {
+    const { status, body } = await api.post(mdmPath, request);
+
+    expect(status).toBe(400);
+    expect(body).toStrictEqual({ error: 'Bad Request', message: 'Header "govUkNotifyKey" is required', statusCode: 400 });
+  });
+
+  it.each([
+    {
+      error: 'Bad Request',
+      expectedStatus: 400,
+    },
+    {
+      error: 'Unauthorized',
+      expectedStatus: 401,
+    },
+    {
+      error: 'Forbidden',
+      expectedStatus: 403,
+    },
+  ])('returns a $expectedStatus response if notify client response with status $expectedStatus', async ({ error, expectedStatus }) => {
+    jest.mocked(sendEmailMethodMock).mockImplementation(() => Promise.reject(generateNotifyError(expectedStatus, errorMessage)));
+
+    const { status, body } = await api.post(mdmPath, request, { govUkNotifyKey });
+
+    expect(status).toBe(expectedStatus);
+    expect(body).toMatchObject({
+      error,
+      message: expect.arrayContaining([errorMessage]),
+      statusCode: expectedStatus,
+    });
+  });
+
+  it('returns a 500 response if notify client response with status 500', async () => {
+    jest.mocked(sendEmailMethodMock).mockImplementation(() => Promise.reject(generateNotifyError(500)));
+
+    const { status, body } = await api.post(mdmPath, request, { govUkNotifyKey });
+
+    expect(status).toBe(500);
+    expect(body).toStrictEqual({
+      statusCode: 500,
+      message: 'Internal server error',
+    });
+  });
+
+  it(`returns a 400 response for empty payload`, async () => {
+    const payload = '';
+    const { status, body } = await api.post(mdmPath, payload, { govUkNotifyKey });
+
+    expect(status).toBe(400);
+    expect(body.error).toMatch('Bad Request');
+    expect(body.message).toMatch('Validation failed (parsable array expected)');
+  });
+
+  it(`returns a 400 response for empty array`, async () => {
+    const payload = [];
+    const { status, body } = await api.post(mdmPath, payload, { govUkNotifyKey });
+
+    expect(status).toBe(400);
+    expect(body.error).toMatch('Bad Request');
+    expect(body.message).toMatch('Request payload is empty');
+  });
+
+  it(`returns a 400 response for broken json`, async () => {
+    const payload = '[]';
+    const { status, body } = await api.post(mdmPath, payload, { govUkNotifyKey });
+
+    expect(status).toBe(400);
+    expect(body.error).toMatch('Bad Request');
+    expect(body.message).toMatch('Validation failed (parsable array expected)');
+  });
```

Comment:
Can we add a test for no GOV.UK Notify response please?

Reply:
Thanks,
Added new test `returns a 422 response if the Notify client responds with empty response`.

Why 422? because it was like this in DTFS.